### PR TITLE
[Feature] Download the Queried output of MV as a CSV

### PR DESCRIPTION
**Pull request by KDwevedi**
closes https://github.com/BharatSahAIyak/admin/issues/158
Additions:
Following bool flags in `GetMaterialViewRequestBody`
- download
- stream

Passed Res Obj from `/mv-data` controller

Returns a CSV when 
```json 
{"download": true}
```
```js
`debugging_view_${new Date().toLocaleString()}.csv`
```

Added field `cols` in the DTO

```js
@ApiProperty({
    description: 'columns for which should be returned in the response',
    example: {
        "cols": ["e_timestamp", "botId"]
    }
})
@IsOptional()
cols: string[] = ['*']
```

**Note:** Since this is a P0 Feature requirement for Samagra Bot, streaming is left `NOT IMPLEMENTED` since debugging isn't complete there.

Will continue to debug and push that separately

### Testing:
Only Manual Testing has been done

Screenshot 2024-08-05 at 2 48 25 PM

## Summary by CodeRabbit

- **New Features**
	- Introduced a new Metrics module to enhance metrics tracking and reporting.
	- Added endpoints for submitting and retrieving metrics data.
	- Enhanced request body with options for CSV download and streaming.
	- Implemented dynamic query generation for metrics requests.

- **Bug Fixes**
	- Improved clarity of import statements within the health service.

- **Documentation**
	- Updated structure and functionality descriptions for new and modified components.

### CODE REVIEW

File: src/metrics.v2/metrics.v2.service.ts [hunk 1]
```diff
@@ -69,6 +76,28 @@ export class MetricsV2Service {
 		params["offset"] = offset;
 
 		const query = selectClause + whereClause + orderClause + limiters;
+
+		if (outputFormat === 'csv') {
+			console.log('[QUERY]', query);
```
Comment (BackendBananeWala) [inline src/metrics.v2/metrics.v2.service.ts:81]: remove consoles

File: src/metrics.v2/metrics.v2.service.ts [hunk 2]
```diff
@@ -69,6 +76,28 @@ export class MetricsV2Service {
...
+			  format: 'CSV'
+			});
+		
+			if (stream) {
+			//   result.stream().pipe(res);
+			res.status(501).send('Streaming not implemented');
+
+			} else {
+			  const data = await result.text();
+			  res.header('Content-Disposition', 'attachment; filename="debugging_data.csv"');
```
Comment (BackendBananeWala) [inline src/metrics.v2/metrics.v2.service.ts:96]: create a filename : {material-view-name}_{timestamp}.csv

Comment (KDwevedi) [reply src/metrics.v2/metrics.v2.service.ts:96]: The person downloading the file is probably not familiar with the MV, MV name probably doesn't mean anything.
I think it should be something like `debugging_view_{timestamp}.csv`.

**Comment by KDwevedi** [comment]
@BackendBananeWala made updates + added a feature, please take a look again

File: src/metrics.v2/metrics.v2.service.ts (hunk 2 above)
Comment (BackendBananeWala) [reply src/metrics.v2/metrics.v2.service.ts]: Sounds good

**Review by BackendBananeWala — APPROVED** [review]

**Pull request merged**